### PR TITLE
windows: enabled Kinesis Firehose, Kinesis Stream, and AWS Filter

### PR DESCRIPTION
**Pull request by rawahars**
## Summary

Currently, Windows release artifacts do not include the following-
- Kinesis Firehose output plugin
- Kinesis Streams output plugin
- AWS Filter

In this change, we are making the required changes to enable support for the same.

The changes for the same include-
- Changing the struct name for Kinesis Firehose plugin so that the build can succeed (`error C2011: 'event': 'struct' type redefinition`)
- Addition of FLB_OUT_KINESIS_STREAMS to the Windows cmake file
- Adding of FLB_FILTER_AWS to the Windows cmake file

----
Enter `[N/A]` in the box, if an item is not applicable to your change.

**Testing**
Before we can approve your change; please submit the following in a comment:
- [ ] Example configuration file for the change
- [ ] Debug log output from testing the change

- [ ] Attached [Valgrind](https://valgrind.org/docs/manual/quick-start.html) output that shows no leaks or memory corruption was found

If this is a change to packaging of containers or native binaries then please confirm it works for all targets.
- [ ] Attached [local packaging test](./packaging/local-build-all.sh) output showing all targets (including any new ones) build.

**Documentation**

- [N/A] Documentation required for this feature

**Backporting**

- [N/A] Backport to latest stable release.

----

Fluent Bit is licensed under Apache 2.0, by submitting this pull request I understand that this code will be released under the terms of that license.

### CODE REVIEW

**Comment by rawahars** [comment]
Tested out all three plugins for Windows using the configuration mentioned in the official documentation.

- https://docs.fluentbit.io/manual/pipeline/filters/aws-metadata
- https://docs.fluentbit.io/manual/pipeline/outputs/firehose
- https://docs.fluentbit.io/manual/pipeline/outputs/kinesis

**Review by PettitWesley — APPROVED** [review]

**Review by PettitWesley — DISMISSED** [review]

File: plugins/out_kinesis_firehose/firehose.h [hunk 1]
```diff
@@ -41,7 +41,7 @@ struct flush {
     size_t data_size;
 
     /* log records- each of these has a pointer to their message in tmp_buf */
-    struct event *events;
+    struct firehose_event *events;
```
Comment (jterry75) [inline plugins/out_kinesis_firehose/firehose.h:44]: Same comments as merge to master review. Is this not a breaking change?

Comment (PettitWesley) [reply plugins/out_kinesis_firehose/firehose.h:44]: Its just renaming an internal data structure so the name won't conflict with other structures in the code base i think, as long as it builds its good

Comment (jterry75) [reply plugins/out_kinesis_firehose/firehose.h:44]: Sounds good

**Review by jterry75 — DISMISSED** [review]

**Review by PettitWesley — APPROVED** [review]

**Review by PettitWesley — APPROVED** [review]

**Review by jterry75 — APPROVED** [review]

**Pull request merged**